src/components/Buttons/Button/Button.tsx

```diff
@@ -39,17 +39,25 @@ export const Button = forwardRef<HTMLButtonElement | HTMLAnchorElement, Props>(
             children,
             disabled = false,
             isBlock = false,
-            isInline = false,
             isLoading = false,
-            isLink = false,
-            isPrimary = false,
             type = 'button',
             href,
-            size = 'normal',
+            size = 'medium',
+            variant = 'filled',
+            context = 'secondary',
             ...rest
         },
         ref
     ) => {
+        const buttonRef = React.useRef<HTMLButtonElement>();
+        const [buttonWidth, setButtonWidth] = React.useState<number>();
+        // figure out width of button, and add that width as styling
+        React.useLayoutEffect(() => {
+            if (buttonRef && buttonRef.current) {
+                setButtonWidth(buttonRef.current.offsetWidth);
+            }
+        }, [buttonRef]);
```

Comment:
In this case, we only measure the element's width during the initial phase. If the width or text changes during the button's lifecycle, it could result in an incorrect width. For example, if we switch the language and all the text changes, the button may have the wrong width. Although this might be an edge case, what do you think about measuring the element's width on demand—specifically when the loading state is true? This way, we will always have the correct width for both the button and the loading state:
```
....
        const buttonRef = React.useRef<HTMLButtonElement | HTMLAnchorElement>(null);
        const [buttonWidth, setButtonWidth] = React.useState<number | undefined>();
        const prevIsLoadingRef = React.useRef(isLoading);

        if (buttonRef.current && isLoading && !buttonWidth) {
            setButtonWidth(buttonRef.current.offsetWidth);
        }

        if (!isLoading && buttonWidth) {
            setButtonWidth(undefined); // Reset to auto width when not loading
        }

        if (typeof children !== 'number' && !children) {
            return null;
        }
...
style={{ width: buttonWidth ? `${buttonWidth}px` : 'auto' }}
....
```

Reply:
I still think we don't need `React.useLayoutEffect` for calculation width.